### PR TITLE
clock

### PR DESCRIPTION
**Pull request by Layrovell**
[DEMO LINK](https://Layrovell.github.io/react_clock/)

### CODE REVIEW

**Review by zhenya-mezhueva — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,66 @@
...
+  }
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
+  setRandomName = () => {
+    this.setState({ clockName: new Date().toLocaleTimeString() });
+    // eslint-disable-next-line no-console
+    console.log(`
+      The Clock was renamed from oddName to
```
Comment (aliceblunt8) [inline src/App.js:24]:
```suggestion
      The Clock was renamed from oldName to
```

File: src/App.js [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {isClockVisible ? (<Clock name={this.state.clockName} />)
+          : <p className="App_offer">cookies?</p>}
```
Comment (aliceblunt8) [inline src/App.js:37]:
```suggestion
        {isClockVisible ? (
        <Clock name={this.state.clockName} />
        ) : (
        <p className="App_offer">cookies?</p>
        )}
```

File: src/Clock/Clock.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      this.setState({ date: new Date().toLocaleTimeString() });
+      // eslint-disable-next-line no-console
+      console.log(new Date().toLocaleTimeString());
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps, prevState, snapshot) {
+  }
```
Comment (aliceblunt8) [inline src/Clock/Clock.jsx:19]: empty method? delete or use it

File: src/App.js [hunk 4]
```diff
@@ -1,25 +1,66 @@
 import React from 'react';
-
 import './App.scss';
+import { Button } from '@material-ui/core';
+import Clock from './Clock/Clock';
+
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: '00:00:00',
```
Comment (aliceblunt8) [inline src/App.js:9]:
```suggestion
    clockName: null,
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 4 above)
Comment (Layrovell) [reply src/App.js:9]: https://prnt.sc/wtgqw3

File: src/App.js [hunk 5]
```diff
@@ -1,25 +1,69 @@
...
-    </div>
-  );
-};
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {isClockVisible ? (
+          <Clock name={this.state.clockName} />
+        ) : (
+          <p className="App_offer">cookies?</p>
```
Comment (varseniuk) [inline src/App.js:39]:
```suggestion
          <p className="App__offer">cookies?</p>
```

File: src/Clock/Clock.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        <p className="clock__time">{date}</p>
+        {name === null
+          ? ''
+          : (
+            <div className="clock__title">
+              The Clock was renamed from oldName to:
+              <p className="clock__text">{`${name}`}</p>
+            </div>
+          )
+        }
```
Comment (varseniuk) [inline src/Clock/Clock.jsx:38]:
```suggestion
        {name && (
            <div className="clock__title">
              The Clock was renamed from oldName to:
              <p className="clock__text">{`${name}`}</p>
            </div>
          )
        }
```

**Review by varseniuk — CHANGES_REQUESTED** [review]

**Review by varseniuk — APPROVED** [review]

**Pull request closed without merging**